mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -43,6 +55,379 @@ struct CompactBlockRow {
+        false
+    }
+
+    pub async fn count_connections(&self) -> usize { self.connections.lock().await.len() }
+}
+
+/// Attempts to process the request (parse url, etc), build up the config and create new lightwalletd connection
+#[cfg(not(target_arch = "wasm32"))]
+#[allow(dead_code)]
+fn spawn_connect(
+    url: &String,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+) -> Result<LightwalletdConnection<Channel>, String> {
+    let uri: Uri = try_s!(url.parse());
+    uri.host().ok_or(ERRL!("Couldn't retrieve host from addr {}", url))?;
+
+    Ok(lightwalletd_connect(url.clone(), event_handlers))
+}
+
+/// Builds up the lightwalletd connection, spawns endless loop that attempts to reconnect to the server
+/// in case of connection errors
+#[allow(dead_code)]
+fn lightwalletd_connect(
+    addr: String,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+) -> LightwalletdConnection<Channel> {
+    let (shutdown_tx, shutdown_rx) = oneshot::channel::<()>();
+    // todo check working with proto format
+    let responses = Arc::new(AsyncMutex::new(JsonRpcPendingRequests::default()));
+    let rpc_client = Arc::new(AsyncMutex::new(None));
+
+    let connect_loop = connect_loop(addr.clone(), responses.clone(), rpc_client.clone(), event_handlers);
+
+    let connect_loop = select_func(connect_loop.boxed(), shutdown_rx.compat());
+    spawn(connect_loop.map(|_| ()));
+    LightwalletdConnection {
+        addr,
+        rpc_client,
+        shutdown_tx: Some(shutdown_tx),
+        responses,
+        protocol_version: AsyncMutex::new(None),
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+#[allow(dead_code)]
+async fn connect_loop<T>(
+    addr: String,
+    _responses: JsonRpcPendingRequestsShared,
+    connection_rpc_client: Arc<AsyncMutex<Option<mpsc::Sender<CompactTxStreamerClient<T>>>>>,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+) -> Result<(), ()> {
+    let delay = Arc::new(AtomicU64::new(0));
+
+    loop {
+        let current_delay = delay.load(AtomicOrdering::Relaxed);
+        if current_delay > 0 {
+            Timer::sleep(current_delay as f64).await;
+        };
+        let uri = try_loop!(Uri::from_str(addr.as_str()), addr, delay);
+
+        let tonic_channel = try_loop!(
+            Channel::builder(uri)
+                .tls_config(ClientTlsConfig::new())
+                .unwrap()
+                .connect()
+                .await,
+            addr,
+            delay
+        );
+        let _client = CompactTxStreamerClient::new(tonic_channel);
+        info!("Light client connected to {}", addr);
+        try_loop!(event_handlers.on_connected(addr.clone()), addr, delay);
+        let last_chunk = Arc::new(AtomicU64::new(now_ms()));
+        let _last_chunk_f = light_last_chunk_loop(last_chunk.clone()).boxed().fuse();
+
+        let (tx, _rx) = mpsc::channel(0);
+        // TODO change event_handlers
+        *connection_rpc_client.lock().await = Some(tx);
+        unimplemented!()
```

Comment:
Unimplemented? Am I correct that you're going to finish it at the next iteration?

Reply:
Yeap, every structure or function with tag `#[allow(dead_code)]` in z_rpc.rs relates to the development of `multi server support`, which will be finished in next sprint.